editor.planx.uk/src/@planx/components/shared/index.ts

```diff
@@ -32,6 +35,33 @@ export const parseBaseNodeData = (
     data?.areTemplatedNodeInstructionsRequired,
 });
 
+/**
+ * Yup validation schema describing BaseNodeData fields
+ */
+export const baseNodeDataValidationSchema: SchemaOf<BaseNodeData> = object({
+  // NodeTags
+  tags: array()
+    .of(
+      string()
+        .oneOf([...NODE_TAGS], "Invalid tag")
+        .required("Tag cannot be empty"),
+    )
+    .optional()
+    .default([]),
+
+  // TemplatedNodeData
+  isTemplatedNode: boolean().optional(),
+  templatedNodeInstructions: string().optional(),
```

Comment:
Something I didn't quite figure out when introducing these is how to have a tighter type - as props each technically has to be introduced optional for existing flow content to work, but ideally we want `templatedNodeInstructions` & `areTemplatedNodeInstructionsRequired` to be required anytime `isTemplatedNode` is `true`. Would be nice to pick up in future :bookmark:

Reply:
Also - I have to admit - I don't totally understand when it's necessary to use `.optional()` versus when optional is simply implied by omission of `.required()` !

Reply:
> _Also - I have to admit - I don't totally understand when it's necessary to use `.optional()` versus when optional is simply implied by omission of `.required()` !_

This is just me being inconsistent sorry! In Yup everything is `.optional()` by default and adding it functionally changes nothing. You have to explicitly set `.required()` when needed.